engine/swarm/stack-deploy.md

```diff
@@ -0,0 +1,242 @@
+---
+description: How to deploy a stack to a swarm
+keywords:
+- guide, swarm mode, composefile, stack, compose, deploy
+title: Deploy a stack to a swarm
+---
+
+When running Docker Engine in swarm mode, you can use `docker stack deploy` to
+deploy a complete application stack to the swarm. The `deploy` command accepts
+a stack description in the form of a [Compose file](/compose/compose-file.md).
+
+The `docker stack deploy` command supports any Compose file of version "3.0" or
+above. If you have an older version, upgrading is simple - see the
+[guide](/compose/compose-file.md#upgrading).
+
+To run through this tutorial, you will need:
+
+1.  A Docker Engine of version 1.13.0 or later, running in [swarm
+    mode](/engine/swarm/swarm-mode.md). If you're not familiar with swarm mode,
+    you might want to read [Swarm mode key concepts ](key-concepts.md) and
+    [How services work](how-swarm-mode- works/services.md).
+
+    > **Note:** If you're trying things out on a local development environment,
+    > you can put your engine into swarm mode with `docker swarm init`.
+    >
+    > If you've already got a multi-node swarm running, keep in mind that all
+    > `docker stack` and `docker service` commands must be run from a manager
+    > node.
+
+2.  [Docker Compose](/compose/install.md) version 1.10 or later.
+
+
+## Set up a Docker registry
+
+Because a swarm consists of multiple Docker Engines, a registry is required to
+distribute images to all of them. You can use the
+[Docker Hub](https://hub.docker.com), or maintain your own, but for the purposes
+of this tutorial it's sufficient to create a throwaway registry.
```

Comment:
change "it's sufficient to create a throwaway registry" to something like "This tutorials creates a simple registry, which you can discard afterward."